starter-code/src/viking.js

```diff
@@ -1,11 +1,111 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health;
+        this.strength = strength;
+    }
+
+    attack() {
+        return this.strength;
+    }
+
+    receiveDamage(damage) {
+        this.health -= damage;
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength) {
+        super(health, strength);
+        this.name = name;
+    }
+
+    attack() {
+        return this.strength;
+    }
+
+    receiveDamage(damage) {
+        this.health -= damage;
+        if (this.health > 0) {
+            return `${this.name} has received ${damage} points of damage`;
+        } else {
+            return `${this.name} has died in act of combat`;
+        }
+    }
+
+    battleCry() {
+        return "Odin Owns You All!"
+    }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    constructor(health, strength) {
+        super(health, strength);
+    }
+
+    attack() {
+        return this.strength;
+    }
+
+    receiveDamage(damage) {
+        this.health -= damage;
+        if (this.health > 0) {
+            return `A Saxon has received ${damage} points of damage`;
+        } else {
+            return "A Saxon has died in combat";
+        }
+    }
+}
 
 // War
-class War {}
+
+class War {
+    constructor() {
+        this.vikingArmy = [];
+        this.saxonArmy = [];
+    }
+
+    addViking(viking) {
+        this.vikingArmy.push(viking);
+    }
+
+    addSaxon(saxon) {
+        this.saxonArmy.push(saxon);
+    }
+
+    vikingAttack() {
+        let randomSaxonNum = Math.floor(Math.random() * this.saxonArmy.length);
+        let randomSaxon = this.saxonArmy[randomSaxonNum];
+        let randomVikingNum = Math.floor(Math.random() * this.vikingArmy.length);
+        let randomViking = this.vikingArmy[randomVikingNum];
+
+        if (randomViking.strength >= randomSaxon.health) {
```

Comment:
Normally here you are supposed to call receiveDamage(damage) before removing the dead solider form the array...
To do that you could hold your function inside a variable (that will at the same time call your function and save the result) and then return this variable at the end.